### ISSUE TITLE
Reorganize code layout to use that from ICON4Py

### ISSUE THREAD

**Issue by jdahm** [GridTools/gt4py#1040]
For example something like this:

```
eve/
  setup.py
  src/
    eve/
  tests/
cartesian/
  setup.py
  src/
    gt4py/
      cartesian/
        gtc/
        gt4py/
  tests/
declarative/ -- in the future once merged
```

**Comment by DropD** [comment]
Important detail (found in icon4py): For precommit mypy to still run on everything it has to be run with the `MYPYPATH` variable set with all the `/src` dirs in it and with the `--explicit-package-bases` option. Otherwise it will not be able to follow the internal namespace package imports.

**Comment by havogt** [comment]
We implemented a simpler layout in #1085
